Improve schedule documentation (#37104)
home-assistant/home-assistant.github.io: source/_integrations/schedule.markdown

ha_integration_type: helper
---

> The **Schedule** {% term integration %} provides a way to create a weekly schedule {% term entity %} in Home Assistant, consisting of time blocks with defined start and end times. The schedule activates when a time block starts and deactivates when it ends, allowing it to be used for triggering or making decisions in automations and scripts.

Reviewer: ```suggestion
The **Schedule** {% term integration %} provides a way to create a weekly schedule {% term entity %} in Home Assistant, consisting of time blocks with defined start and end times. The schedule is active when a time block starts and becomes inactive when it ends, allowing it to be used for triggering or making decisions in automations and scripts.
```

Schedule blocks:
  description: >
    Press and drag to select time blocks for each day of the week.
    It is not possible to create overlapping time blocks on the same day.
{% endconfiguration_basic %}

After creating schedule blocks, you can press a block to edit the details.

{% configuration_basic %}
Start:
  required: true
  type: time
  description: The start time to mark the schedule as active/on.
End:
  required: true
  type: time
  description: The end time to mark as inactive/off again.
Additional data:
  required: false
  type: map
  description: Additional data should be a mapping of attribute names to values, which will be added to the entity's attributes when the block is active.
{% endconfiguration_basic %}

The **Additional data** entry of each block should be a mapping of attribute names to values.
For example, a block with the following data will show `brightness` and `color_temp` as {% term entity %} attributes when the block is active:

```yaml
brightness: 100
color_temp: 4000
```

## YAML configuration

Alternatively, this {% term integration %} can be configured and set up manually via YAML instead.

> To enable the Integration sensor in your installation, add the following to your {% term "`configuration.yaml`" %} file.

Reviewer: This still doesn't specifically say what should be added.  You may be referring to the yaml below, but that's not clear. WDYT about the suggestion below: 

```
Alternatively, this integration can be set up through {% term "configuration.yaml" %}. 

(Can we move the configuration.yaml explanation here, then move the example below that?)

### Example `light_schedule`
(Example here)

{% note %} Additional data can be added using the `data` block for each schedule block.  {% endnote %}
```

Author: It is the default HA style to first have a YAML example and then the configuration block, so I would keep it that way.

ha_integration_type: helper
---

> The **Schedule** {% term integration %} provides a way to create a weekly schedule {% term entity %} in Home Assistant that can be used to trigger or make decisions in your automations and scripts.

Reviewer: We're going straight into time blocks a little bit more below, it would be great to explain in this introduction that a schedule consists of time blocks: 

```suggestion
The **Schedule** {% term integration %} provides a way to create a weekly schedule {% term entity %} in Home Assistant, consisting of time blocks with defined start and end times. The schedule activates when a time block starts and deactivates when it ends, allowing it to be used for triggering or making decisions in automations and scripts.
```

{% include integrations/config_flow.md %}

{% configuration_basic %}
Name:

>   description: Friendly name of the schedule. It can be changed again later.

Reviewer: ```suggestion
  description: Friendly name of the schedule. 
```

Icon:
  description: Icon to display in the frontend for this schedule.
Schedule blocks:
  description: >

>     Drag and drop with your mouse to select time boxes for each day of the week.
>     It is not possible to create overlapping time boxes on the same day.
> {% endconfiguration_basic %}

Reviewer: ```suggestion
    Press and drag to select time blocks for each day of the week.
    It is not possible to create overlapping time blocks on the same day.
{% endconfiguration_basic %}
```

> After you have created schedule blocks, you can use a **single left click** on a block to open an edit view for this time block.

Reviewer: ```suggestion
After creating schedule blocks, you can press a block to edit the details.
```

> {% configuration_basic  %}

Reviewer: ```suggestion
{% configuration_basic %}
```

Start:
  required: true
  type: time
  description: The start time to mark the schedule as active/on.
End:
  required: true
  type: time
  description: The end time to mark as inactive/off again.
Additional data:
  required: false
  type: map

>   description: Additional data to add to the entity's attributes when this block is active.

Reviewer: ```suggestion
  description: Additional data should be a mapping of attribute names to values, which will be added to the entity's attributes when the block is active.
```

{% endconfiguration_basic %}

The **Additional data** entry of each block should be a mapping of attribute names to values.
For example, a block with the following data will show `brightness` and `color_temp` as {% term entity %} attributes when the block is active:

```yaml
brightness: 100
color_temp: 4000
```

## YAML configuration

Alternatively, this {% term integration %} can be configured and set up manually via YAML instead.

> To enable the Integration sensor in your installation, add the following to your {% term "`configuration.yaml`" %} file:

Reviewer: I think something is missing here, or the goal of this sentence is pretty unclear.

Author: Deleted a little bit too much/too little.

          default: {}
{% endconfiguration %}

## Attributes

A schedule entity exports state attributes that can be useful in automations and templates.

| Attribute | Description |
| ----- | ----- |
| `next_event` | A datetime object containing the next time the schedule is going to change state. |
| `key_1`, `key_2`, ... | The mapping values from **Additional data** / `data` settings of a time block when the respective block is active. |

## Automation example

A schedule creates an on/off (schedule) sensor within the times set.

> By using the `light_schedule` example from above in an automation, we can turn on a light only during the time box when a schedule is active.

Reviewer: ```suggestion
By incorporating the `light_schedule` example from above in an automation, we can turn on a light when the schedule is active.
```

{% raw %}

```yaml
triggers:
  - trigger: state
    entity_id:
      - schedule.light_schedule
    to: "on"
actions:
  - action: light.turn_on
    target:
      entity_id: light.kitchen
    data:
      brightness_pct: "{{ state_attr('schedule.light_schedule', 'brightness') }}"
      kelvin: "{{ state_attr('schedule.light_schedule', 'temperature') }}"
```

{% endraw %}

> The same schedule can be used to turn the lights off again once the time blocks of the schedule have passed:

Reviewer: ```suggestion
Another automation can be added to turn the lights off once the schedule is inactive: 
```

          default: {}
{% endconfiguration %}

## Attributes

> A schedule entity's state exports attributes that can be useful in automations and templates.

Reviewer: ```suggestion
A schedule entity exports state attributes that can be useful in automations and templates.
```

| Attribute | Description |
| ----- | ----- |
| `next_event` | A datetime object containing the next time the schedule is going to change state. |

> | Additional data | The entries in **Additional data** / `data` of a schedule block when the respective block is active. |

Reviewer: Use the attribute key here, then use the description to re-mention that we're talking about addtional data.

Author: I concur here. Is this better?

## Automation example

A schedule creates an on/off (schedule) sensor within the times set.

> Using the `light_schedule` example from above in an automation might look like this:

Reviewer: This should briefly explain what the example below does. When the schedule is activated, it should turn on the lights. 

The light will only be turned on when the schedule is activated.
